synapse/handlers/device.py

```diff
@@ -758,12 +758,13 @@ async def rehydrate_device(
 
         # If the dehydrated device was successfully deleted (the device ID
         # matched the stored dehydrated device), then modify the access
-        # token to use the dehydrated device's ID and copy the old device
-        # display name to the dehydrated device, and destroy the old device
-        # ID
+        # token and refresh token to use the dehydrated device's ID and
+        # copy the old device display name to the dehydrated device,
+        # and destroy the old device ID
         old_device_id = await self.store.set_device_for_access_token(
             access_token, device_id
         )
+        await self.store.move_device_refresh_token(old_device_id, device_id)
```

Comment:
Probably worth following a similar naming scheme?

```suggestion
        await self.store.set_device_for_refresh_token(old_device_id, device_id)
```